api/client/image/build.go

```diff
@@ -211,6 +227,9 @@ func runBuild(dockerCli *client.DockerCli, options buildOptions) error {
 
 	// Setup an upload progress bar
 	progressOutput := streamformatter.NewStreamFormatter().NewProgressOutput(progBuff, true)
+	if !dockerCli.IsTerminalOut() {
```

Comment:
@tonistiigi `progressOutput` was reused as `lastProgressOutput` is just a wrapper on top. I think either way might be fine.